plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts

```diff
@@ -80,10 +80,13 @@ export function createRouter({
     },
   );
 
-  router.get('/find/selector/:argoAppSelector', async (request, response) => {
-    const argoAppSelector = request.params.argoAppSelector;
-    response.send(await argoSvc.findArgoApp({ selector: argoAppSelector }));
-  });
+  router.get(
+    '/find/selector/:argoAppSelector([^/]*)',
```

Comment:
This regex looks off. In the case of the test data `/find/selector/prefix/name=test-app`, doesn't this call `findArgoApp` with `{ selector: "prefix" }`? I think you want to use `.+` for the regex. You should add a test to ensure this question is answered.

Reply:
I have added extra verifications in the tests to ensure that the URL is parsed correctly when using arbitrary selectors, and that the underlying method is called with the correct input.